### ISSUE TITLE
Search facet requires using a defined type 

### ISSUE THREAD

**Issue by agnellvj** [karmi/retire#225]
So the problem is I need to do a `Statistical` facet but right now there isn't one defined.

I took a peek at the `Facet` class and I think this could work as a workaround by initializing `@value = {}`.  This would allow the following to work:

```
s = Tire.search 'my_index' do
   query do
     string 'title:T*'
   end
   facet 'my_stats', { :statistical => { :field => 'field_name' } }
end
```

The other option is to implement  `Facet#statistical`.  I took a peek at the way to do it and it doesn't seem like it is straightforward because you can either use the `field` option or `script` option with params.

Not sure what the best way to go.  Thoughts?

**Comment by vhyza** [comment]
Hello,

I think better option is to implement `Facet#statistical` method to keep it consistent with another types of facets.

**Comment by karmi** [comment]
Statistical facet was implemented in 8b06760.
